lib/internal/fs/glob.js

```diff
@@ -0,0 +1,370 @@
+        key += '/';
+      }
+    }
+    return key;
+  }
+}
+
+class Glob {
+  #root;
+  #exclude;
+  #cache = new Cache();
+  #results = [];
+  #queue = [];
+  constructor(patterns, options = kEmptyObject) {
+    validateObject(options, 'options');
+    const { exclude, cwd } = options;
+    if (exclude != null) {
+      validateFunction(exclude, 'options.exclude');
+    }
+    this.#root = cwd ?? '.';
+    this.#exclude = exclude;
+    this.matchers = ArrayPrototypeMap(patterns, (pattern) => new (lazyMinimatch().Minimatch)(pattern, {
+      __proto__: null,
+      nocase: isWindows || isOSX,
+      windowsPathsNoEscape: true,
+      nonegate: true,
+      nocomment: true,
+      optimizationLevel: 2,
+      platform: process.platform,
+      nocaseMagicOnly: true,
+    }));
+  }
+
+  globSync() {
+    ArrayPrototypePush(this.#queue, {
+      __proto__: null,
+      path: '.',
+      patterns: ArrayPrototypeFlatMap(this.matchers, (matcher) => ArrayPrototypeMap(matcher.set,
+                                                                                    (pattern, i) => new Pattern(
+                                                                                      pattern,
+                                                                                      matcher.globParts[i],
+                                                                                      new SafeSet([0]),
+                                                                                      new SafeSet(),
+                                                                                    ))),
+    });
+
+    while (this.#queue.length > 0) {
+      const item = ArrayPrototypePop(this.#queue);
+      for (let i = 0; i < item.patterns.length; i++) {
+        this.#addSubpatterns(item.path, item.patterns[i]);
+      }
+      this.#subpatterns
+        .forEach((patterns, path) => ArrayPrototypePush(this.#queue, { __proto__: null, path, patterns }));
+      this.#subpatterns.clear();
+    }
+    return this.#results;
+  }
+  #subpatterns = new SafeMap();
+  #addSubpattern(path, pattern) {
+    if (!this.#subpatterns.has(path)) {
+      this.#subpatterns.set(path, [pattern]);
+    } else {
+      ArrayPrototypePush(this.#subpatterns.get(path), pattern);
+    }
+  }
+  #addSubpatterns(path, pattern) {
+    const seen = this.#cache.add(path, pattern);
+    if (seen) {
+      return;
+    }
+    const fullpath = resolve(this.#root, path);
+    const stat = this.#cache.statSync(fullpath);
+    const last = pattern.last;
+    const isDirectory = stat?.isDirectory() || (stat?.isSymbolicLink() && pattern.hasSeenSymlinks);
+    const isLast = pattern.isLast(isDirectory);
+    const isFirst = pattern.isFirst();
+
+    if (isFirst && isWindows && typeof pattern.at(0) === 'string' && StringPrototypeEndsWith(pattern.at(0), ':')) {
+      // Absolute path, go to root
+      this.#addSubpattern(`${pattern.at(0)}\\`, pattern.child(new SafeSet([1])));
```

Comment:
all these .at’s need to be primordials

Reply:
it is an internal method
https://github.com/MoLow/node/blob/11f657cc08ad4d93c7fc25cf89825d822f65c8c7/lib/internal/fs/glob.js#L111-L113